File: regions/system/library/dart/featureflag/lib/src/flag.dart
class FlagVariant {
  final String name;
  final String value;
  final double weight;

  const FlagVariant({
    required this.name,
    required this.value,
    this.weight = 1.0,
  });
}

class FeatureFlag {
  final String id;
  final String flagKey;
  final String description;
  final bool enabled;
  final List<FlagVariant> variants;

  const FeatureFlag({
    required this.id,
    required this.flagKey,
    required this.description,
    required this.enabled,
    this.variants = const [],
  });
}

File: regions/system/library/rust/featureflag/src/flag.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct FlagVariant {
    pub name: String,
    pub value: String,
    pub weight: i32,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct FeatureFlag {
    pub id: String,
    pub flag_key: String,
    pub description: String,
    pub enabled: bool,
    pub variants: Vec<FlagVariant>,
}
